elixir/apps/api/lib/api/device/channel.ex

```diff
@@ -80,15 +82,23 @@ defmodule API.Device.Channel do
   end
 
   @impl true
-  def handle_in("list_relays", %{"resource_id" => resource_id}, socket) do
+  def handle_in("prepare_connection", %{"resource_id" => resource_id} = attrs, socket) do
+    connected_gateway_ids = Map.get(attrs, "connected_gateway_ids", [])
```

Comment:
@conectado `connected_gateway_ids` is an optional param now, but would be great if you send it - we will reuse existing gateway connections instead of establishing a connection to a random one.